atc/atccmd/command.go

```diff
@@ -994,17 +999,36 @@ func (tripper mitmRoundTripper) RoundTrip(req *http.Request) (*http.Response, er
 	return tripper.RoundTripper.RoundTrip(req)
 }
 
-func (cmd *RunCommand) tlsConfig() (*tls.Config, error) {
+func (cmd *RunCommand) tlsConfig(logger lager.Logger, dbConn db.Conn) (*tls.Config, error) {
 	var tlsConfig *tls.Config
+	tlsConfig = atc.DefaultTLSConfig()
 
 	if cmd.isTLSEnabled() {
-		cert, err := tls.LoadX509KeyPair(string(cmd.TLSCert), string(cmd.TLSKey))
-		if err != nil {
-			return nil, err
-		}
+		tlsLogger := logger.Session("tls-enabled")
+		if cmd.EnableAutocert {
+			tlsLogger.Info("starting autocert manager")
 
-		tlsConfig = atc.DefaultTLSConfig()
-		tlsConfig.Certificates = []tls.Certificate{cert}
+			cache, err := newDbCache(dbConn)
+			if err != nil {
+				return nil, err
+			}
+			m := autocert.Manager{
+				Prompt:     autocert.AcceptTOS,
+				Cache:      cache,
+				HostPolicy: autocert.HostWhitelist(cmd.ExternalURL.URL.Hostname()),
+				Client:     &acme.Client{DirectoryURL: cmd.ACMEURL.URL.String()},
+				ForceRSA:   true,
```

Comment:
nit: looks like this is deprecated and ignored now: https://godoc.org/golang.org/x/crypto/acme/autocert#Manager